pkg/cmd/roachtest/tests/connection_latency.go

```diff
@@ -102,15 +106,78 @@ func runConnectionLatencyTest(
 		cockroachUsWest := loadGroups[1].loadNodes
 		cockroachEuWest := loadGroups[2].loadNodes
 
+		_, cleanup := setupPrometheusConnectionLatency(ctx, t, c, loadGroups.loadNodes())
+		defer cleanup()
+
 		runWorkload(loadGroups[0].roachNodes, cockroachUsEast, regionUsEast)
 		runWorkload(loadGroups[1].roachNodes, cockroachUsWest, regionUsWest)
 		runWorkload(loadGroups[2].roachNodes, cockroachEuWest, regionEuWest)
 	} else {
+		_, cleanup := setupPrometheusConnectionLatency(ctx, t, c, c.Node(c.Spec().NodeCount))
+		defer cleanup()
 		// Run only on the load node.
 		runWorkload(c.Range(1, numNodes), c.Node(numNodes+1), regionUsCentral)
 	}
 }
 
+// setupPrometheusConnectionLatency initializes prometheus to run against a new
+// PrometheusConfig. It creates a prometheus scraper for all load nodes.
+// Returns the created PrometheusConfig if prometheus is initialized, as well
+// as a cleanup function which should be called in a defer statement.
+func setupPrometheusConnectionLatency(
```

Comment:
when you run this, can you go on `<external_ip>:9090` and verify that the metrics exist?

Reply:
well it seemed to work since i looked in my artifacts and found some prometheus metrics. but i don't really know how to interpret them! probably because i don't know PromQL. could you help me figure out how i would view the p95 and p99 latencies of the connections in each region?

[8.prometheus-snapshot.tar.gz](https://github.com/cockroachdb/cockroach/files/7187676/8.prometheus-snapshot.tar.gz)

Reply:
the metric doesn't seem to render?
`histogram_quantile(0.95, rate(workload_connectionlatency_connect_from_us_east1_b_to_us_east1_b_duration_seconds_bucket[5m]))`

would find the percentile. unfortunately it doesn't seem like any metrics seem to have been registered but autocomplete seems to have succeeded - did any connections take place?

see also: https://www.robustperception.io/how-does-a-prometheus-histogram-work

Reply:
i do see data. maybe make sure the time window is correct?

<img width="1663" alt="Screen Shot 2021-09-20 at 2 17 19 AM" src="https://user-images.githubusercontent.com/1320573/133962859-972db1bd-4c4d-423c-aed0-aa6ff50e4941.png">


also i tried this to try to get name pattern matching: `histogram_quantile(0.95, {__name__=~"workload_connectionlatency_connect_from_.*_bucket"})`

Reply:
hmm i tried a large window. oh well, glad to know it works. lgtm!